repl/src/main/scala/ammonite/repl/interp/Interpreter.scala

```diff
@@ -454,16 +493,65 @@ class Interpreter(prompt0: Ref[String],
 
       def exec(file: Path): Unit = apply(read(file))
 
-      def module(file: Path): Unit = {
+      def loadModule(file: Path, storage: Storage, cacheTag: String): Unit ={
+        val code = read(file)
         val (pkg, wrapper) = Util.pathToPackageWrapper(file, wd)
-        processModule(read(file), wrapper, pkg) match{
+        processModule(code, wrapper, pkg) match {
           case Res.Failure(ex, s) => throw new CompilationError(s)
           case Res.Exception(t, s) => throw t
-          case x => //println(x)
+          case Res.Success(data) =>
+            if(scriptCaching) {
+              val files = for (d <- data._2) yield (d._1, d._2)
+              storage.asInstanceOf[Storage.Folder]classFilesListSave(pkg, wrapper, files, cacheTag)
+            }
         }
         init()
       }
 
+      // Try loading script from cache
+      def cachedModule(path: Path) = {
+        Timer.startTime = System.nanoTime()
+
+        val code: String = read(path)
+        val (pkg, wrapper) = Util.pathToPackageWrapper(path, wd)
+        val cacheTag = "cache" + Util.md5Hash(Iterator(code.getBytes))
+          .map("%02x".format(_)).mkString
+        if(!code.contains("load(")) {
```

Comment:
Why do we need to check if the code contains `load(`? 

I don't think this is necessary; we should cache a module regardless of what it does. Even if we did want to do this, it would immediately break once someone called the function via `load (...)` or `load.apply (...)`